### PR TITLE
Removed `SessionIdentifierAwareInterface`

### PR DESCRIPTION
**Pull request by alexmerlin**
|    Q          |   A
|-------------- | ------
| Documentation | yes
| Bugfix        | no
| BC Break      | yes
| New Feature   | no
| RFC           | no
| QA            | no

### Description

As requested in #59 this commit:

- adds `SessionIdentifierAwareInterface`'s `getId()` method to `SessionInterface`
- removes `SessionIdentifierAwareInterface`

### CODE REVIEW

**Comment by alexmerlin** [comment]
@gsteel Here are the changes for the 2.0.x branch.

File: src/SessionInterface.php [hunk 1]
```diff
@@ -68,4 +68,9 @@ public function regenerate(): self;
      * true if the instance was produced via regenerate().
      */
     public function isRegenerated(): bool;
+
+    /**
+     * Retrieve the session identifier.
+     */
+    public function getId(): string;
```
Comment (gsteel) [inline src/SessionInterface.php:77]: It'd be nice if we could improve type precision by introducing `@return non-empty-string`

File: docs/book/persistence.md [hunk 2]
```diff
@@ -71,19 +71,12 @@ session:
...
 identifier internally, and, when it comes time to store the session data,
 persistence implementations can query that method in order to retrieve the
 session identifier.
 
-Considering that persistence implementations also _create_ the session instance,
-we recommend that implementations only create instances that implement the
-`SessionIdentifierAwareInterface` going forward in order to make themselves
-async compatible.
-
 ## Persistent sessions
```
Comment (gsteel) [inline docs/book/persistence.md]: All the changes to the docs need to be reverted.

First of all, we need a patch to `1.x` to move the docs from the root `/docs/` folder to `/docs/v1`, ensuring we set up redirects for `/some-file.md` to `/v1/some-file.md`

Example: https://github.com/laminas/laminas-validator/pull/244

This needs to be released and merged up into `2.x`

Then, we need a patch to `2.x` that duplicates the `v1` docs tree to a `v2` folder:

Example: https://github.com/laminas/laminas-validator/pull/248.

At this point, we'll need to rebase this patch, and then make the required changes to the docs in the v2 tree.

Make sense?

**Review by gsteel — CHANGES_REQUESTED** [review]
Thanks @alexmerlin - I should have mentioned that we'll need to get the docs setup for the v1/v2 split before we can start merging patches that require docs updates.

We'll also need to setup a migration guide too.

File: docs/book/persistence.md (hunk 2 above)
Comment (alexmerlin) [reply docs/book/persistence.md]: Sure, I forgot to make the v2 folder - I'll fix it.

Comment (alexmerlin) [reply docs/book/persistence.md]: See #63 for the `v1` documentation modifications.

**Comment by gsteel** [comment]
There's something weird going on here @alexmerlin - it'll be easier if you send another PR to first duplicate the v1 docs to v2 without changing anything, then, rebase _this_ PR on to the tip of 2.0.x, re-applying the changes to the v2 docs tree.

**Pull request closed without merging**